packages/components/src/__future__/Tooltip/Focusable.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { HTMLAttributes, ReactNode, useRef } from "react"
+import classnames from "classnames"
+import { FocusableOptions, useFocusable } from "react-aria"
+import styles from "./Focusable.module.scss"
+
+export type FocusableProps = {
+  children: ReactNode
+} & FocusableOptions &
+  HTMLAttributes<HTMLDivElement>
+
+// todo this component will be reusable util for all components to give them focusable wrapper
+export const Focusable = ({
+  children,
+  className,
+  ...props
+}: FocusableProps): JSX.Element => {
+  const ref = useRef<HTMLDivElement>(null)
+  const { focusableProps } = useFocusable(props, ref)
+
+  return (
+    <div
+      ref={ref}
+      className={classnames(styles.focusableWrapper, className)}
+      {...focusableProps}
+      data-inline-hidden-content
+      // We want the div to be focusable for keyboard users,
+      // but screen readers will have the VisuallyHidden content
+      // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+      tabIndex={0}
+      // Negate the aria description (added by RAC) as we have the VisuallyHidden content
+      aria-describedby={undefined}
```

Comment:
If I understand this this setting (+ the `inline-hidden-content`) it's to provide visually hidden content rather than rely on `aria-describedby` to describe its contents. Is this always the case though? Or is it more of a tooltip/KAIO thing, eg. would this be applicable to Table Rows, Card or Data point on a graph etc. As a utility I'd expect it to work on anything and everything, even more so if I'm trying to create a bespoke component which no one's ever thought of.

So I think we need to decide, is this a public facing utility or not;
- if so, then it might be better to make these optimisations an opt-in API to cater for both DST use cases and make it easier for consumers (and us) to both understand the utility and document its use cases.
- if not, then this should probably remain an internal utility, and if the consumer needs something like this, they can read RAC docs on how to compose something similar. (in which we need a place to put internal utils...)

Although outside of these two options, I don't quite understand the value of this utility to consumers.

Reply:
yeah that's right, it's used to provide VisuallyHidden content instead of describedBy tooltip. Basically the `inline-hidden-content` on a tooltip (and other) triggers tells the `Tooltip` to render the content also into VisuallyHidden and so the `describedby` isn't needed.

That is quite specific tooltip a11y problem/solution but I'm not sure how it will work for other future components. I can imagine this being reusable, but not sure if there's reason to.